azurerm/resource_arm_network_interface.go

```diff
@@ -356,7 +362,7 @@ func resourceArmNetworkInterfaceDelete(d *schema.ResourceData, meta interface{})
 		defer azureRMUnlockByName(networkSecurityGroupName, networkSecurityGroupResourceName)
 	}
 
-	configs := d.Get("ip_configuration").(*schema.Set).List()
+	configs := d.Get("ip_configuration").([]interface{})
```

Comment:
Correct me if I'm wrong, but I think this will crash with the old state (with Set) without migration.